helix/appendix/alternate-visual-studio-structures.rst

```diff
@@ -0,0 +1,231 @@
+also consider using the `Filtered Solutions <https://docs.microsoft.com/en-us/visualstudio/ide/filtered-solutions?view=vs-2019>`__
+feature of Visual Studio 2019 to reduce the number of projects open when working
+on a particular module.
+
+
+The role of Visual Studio projects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Aside from the pragmatic issues, the architectural question at the root of
+this decision is the role of a project/assembly in Visual Studio. This is
+essentially the subject of Robert C. Martin's *Principles of Package and Component Design*,
+which contain the `underlying principles <package-principles>`__ that Sitecore Helix
+is based on. Based on these principles, Visual Studio projects form a "physical" boundary
+and unit of change for modules in Sitecore Helix.
+
+Martin however makes an assumption that modules (packages) are released / deployed
+independently. Due to the versioning and testing complexity this would put on the typical Sitecore
+solution, Sitecore Helix actually `discourages partial deploys of modules </devops/deployment/strategy>`__,
+meaning that the unit of release is the entirety of the application. If we are not deploying
+and versioning these assemblies independently, it is a valid question as to whether
+the modules really merit their own Visual Studio projects.
+
+The modular architecture practices of Sitecore Helix though are meant to improve solution
+*maintainability*. The convention of using a separate web application
+project for each module (which deploys to Sitecore CM/CD) means that all the code and
+configuration related to a module are located in one place. Combine that with the module
+filesystem and solution folder structure, and you also have the items, tests, and other related
+code for that module. Practically, it's very easy for a future developer to find/discover
+the code related to a desired system change. It's also easier to track and manage the dependencies
+to and from that module.
+
+It is true that `FxCop rules <https://www.teamdevelopmentforsitecore.com/Blog/sitecore-helix-fxcop-rules>`__
+can be used to enforce Sitecore Helix dependency rules without separate projects, effectively giving you a
+"logical" boundary between modules to help enforce dependencies. However the loss of physical *Common Closure*
+and the potential maintenance impact should definitely be considered in taking this approach.
+
+
+How to structure when consolidating
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+How you consolidate the projects in your solution will ultimately depend on
+the issue(s) you are attempting to address, and your view on the purpose and role
+of Visual Studio projects.
+
+* You may group Features into combined projects based on the concept of
+  `Feature Groups </principles/architecture-principles/modules.html#groups>`__. This will
+  reduce the number of projects but care must be taken to avoid dependencies between
+  Features in a group.
```

Comment:
OK, yes -- this still is following the barriers. Do see some folks using the group methodology to "allow" for inter-module dependencies. That's not what we're talking about

Reply:
further clarified this